sunpy/util/metadata.py

```diff
@@ -14,27 +14,66 @@ class MetaDict(OrderedDict):
 
     This class handles everything in lower case. This allows case
     insensitive indexing.
+
+    If the key 'keycomments' exists, its value must be a dictionary mapping
+    keys in the `MetaDict` to their comments. The casing of keys in the
+    keycomments dictionary is not significant. If a key is removed from the
+    `MetaDict`, it will also be removed from the keycomments dictionary.
+    Additionally, any extraneous keycomments will be removed when the
+    `MetaDict` is instantiated.
     """
     def __init__(self, *args):
         """
-        Creates a new MapHeader instance.
+        Creates a new MetaDict instance.
         """
-        # Store all keys as upper-case to allow for case-insensitive indexing
+        # Store all keys as lower-case to allow for case-insensitive indexing
         # OrderedDict can be instantiated from a list of lists or a tuple of tuples
         tags = dict()
         if args:
             args = list(args)
             adict = args[0]
             if isinstance(adict, list) or isinstance(adict, tuple):
-                tags = OrderedDict((k.upper(), v) for k, v in adict)
```

Comment:
These `upper` calls aren't necessary because the keys are converted to lowercase in the `super()` call below.